### PR TITLE
Implement the event-source management service.

### PR DESCRIPTION
**Pull request by wimax-grapl**
# Quick diatribe on naming
I like "event source" or "event source service" over "event source management" or "event source management service." we should probably figure out a standardized way to name these services. I find management to be a lot of typing for not much lexical meaning-add. like, OrganizationService vs OrganizationManagementService? What is OrganizationService for if not managing organizations? 

# Back to usual PR stuff

### Which issue does this PR correspond to?
https://github.com/grapl-security/issue-tracker/issues/847

### What changes does this PR make to Grapl? Why?
1. Build out the Event Source Management.
2. Integration test for it
3. Seriously reduce repeated code in `grapl-local-infra` with regards to postgres dbs !!!

### How were these changes tested?
integration tests added

### CODE REVIEW

File: src/rust/event-source/src/client.rs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        let endpoint = std::env::var(ADDRESS_ENV_VAR).expect(ADDRESS_ENV_VAR);
+
+        HealthcheckClient::wait_until_healthy(
+            endpoint.clone(),
+            Self::SERVICE_NAME,
+            Duration::from_millis(10000),
+            Duration::from_millis(500),
+        )
+        .await
+        .expect("plugin-registry never reported healthy");
```
Comment (colin-grapl) [inline src/rust/event-source/src/client.rs:35]: It seems odd to panic here given that this function returns a Result, but my guess is that you don't want to add something like "never connected" to the error type used in a bunch of places? nbd if you don't want to change

File: src/rust/event-source/src/server.rs [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    config::EventSourceConfig,
+    db::EventSourceDbClient,
+    error::EventSourceError,
+};
+
+pub async fn exec_service(config: EventSourceConfig) -> Result<(), Box<dyn std::error::Error>> {
+    let api_impl = EventSourceApiImpl::try_from(&config).await?;
+    // I haven't quite figured out the right place to put the migrate.
+    // Perhaps when we construct the db client?!
+    api_impl.db_client.migrate().await?;
```
Comment (colin-grapl) [inline src/rust/event-source/src/server.rs:25]: I think you want to use the `migrate!` macro

**Review by colin-grapl — APPROVED** [review]

File: src/rust/event-source/src/server.rs (hunk 2 above)
Comment (wimax-grapl) [reply src/rust/event-source/src/server.rs:25]: look at the implementation of this function

File: src/rust/event-source/src/client.rs (hunk 1 above)
Comment (wimax-grapl) [reply src/rust/event-source/src/client.rs:35]: more importantly, copy-paste fail here :)

**Pull request merged**